e2e/test/scenarios/models/reproductions/29951-model-editor-results-metadata.cy.spec.js

```diff
@@ -25,13 +25,11 @@ describe("issue 29951", () => {
   beforeEach(() => {
     restore();
     cy.signInAsAdmin();
-    cy.intercept("POST", "/api/dataset").as("dataset");
     cy.intercept("PUT", "/api/card/*").as("updateCard");
   });
 
   it("should allow to run the model query after changing custom columns (metabase#29951)", () => {
     cy.createQuestion(questionDetails, { visitQuestion: true });
-    cy.wait("@dataset");
```

Comment:
If `visitQuestion` is set to `true`, it will automatically intercept and wait for the `@dataset` aliased request.
This was the source of this test flakiness - the additional `cy.wait("@dataset")`

To elaborate on this a bit further:
- having multiple `cy.intercept` for the same route does not create an issue
- aliasing multiple identical intercepts with the same route doesn't necessarily have to create an issue

But adding an additional wait can and will create an issue.
In this test, we had four waits for a route that's invoked only three times.